radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/addressbook/AddressBookEntry.java

```diff
@@ -133,11 +134,11 @@ public static AddressBookEntry createBlacklisted(RadixNodeUri uri, Instant black
 		);
 	}
 
-	public static AddressBookEntry create(RadixNodeUri uri, Optional<Instant> lastSuccessfulConnection) {
+	public static AddressBookEntry create(RadixNodeUri uri, Optional<LatestConnectionStatus> latestConnectionStatus) {
 		return new AddressBookEntry(
 			uri.getNodeId(),
 			Optional.empty(),
-			ImmutableSet.of(new AddressBookEntry.PeerAddressEntry(uri, lastSuccessfulConnection, Optional.empty()))
+			ImmutableSet.of(new AddressBookEntry.PeerAddressEntry(uri, latestConnectionStatus, Optional.empty()))
```

Comment:
It is probably gonna fail with NullPointerException if latestConnectionStatus is null. Is that what we want?

Reply:
I think so, it should never be null

Reply:
I believe it is better to enforce that explicitly.

Reply:
so that's a broader question, should we add `Objects.requireNonNull` everywhere where null is not expected?

Reply:
I wouldn't trust recommendations to guarantee safety. I believe it is better to enforce it in the code. It is already implicitly enforced by the ImmutableSet, but I believe it is better to enforce requirements explicitly than relying on implicit checks. If this data structure is changed in the future to one that accepts null and I believe is reasonable to assume that this requirement could be neglected and then the code is unsafe.

Reply:
We agreed to discuss this with the team.